packages/jellyfish-api-core/__tests__/category/icxorderbook/claimDFCHTLC.test.ts

```diff
@@ -0,0 +1,191 @@
+import { ContainerAdapterClient } from '../../container_adapter_client'
+import { MasterNodeRegTestContainer } from '@defichain/testcontainers'
+import {
+  ICXClaimDFCHTLCInfo, ICXDFCHTLCInfo, ICXEXTHTLCInfo, ICXHTLCStatus,
+  ICXHTLCType, ICXListHTLCOptions
+} from '../../../src/category/icxorderbook'
+import { accountDFI, idDFI, accountBTC, ICXSetup, symbolDFI } from './icx_setup'
+import { BigNumber, RpcApiError } from '../../../src'
+
+describe('ICXOrderBook.calimDFCHTLC', () => {
+  const container = new MasterNodeRegTestContainer()
+  const client = new ContainerAdapterClient(container)
+  const icxSetup = new ICXSetup(container, client)
+
+  beforeAll(async () => {
+    await container.start()
+    await container.waitForReady()
+    await container.waitForBlock(1)
+    await icxSetup.createAccounts()
+    await icxSetup.createBTCToken()
+    await icxSetup.initializeTokensIds()
+    await icxSetup.mintBTCtoken(100)
+    await icxSetup.fundAccount(accountDFI, symbolDFI, 500)
+    await icxSetup.fundAccount(accountBTC, symbolDFI, 10) // for fee
+    await icxSetup.createBTCDFIPool()
+    await icxSetup.addLiquidityToBTCDFIPool(1, 100)
+    await icxSetup.setTakerFee(0.001)
+  })
+
+  afterAll(async () => {
+    await container.stop()
+  })
+
+  afterEach(async () => {
+    // enable this after #ain/583
+    // await icxSetup.closeAllOpenOffers()
+  })
+
+  it('should claim DFC HTLC for DFI sell order', async () => {
+    const { createOrderTxId } = await icxSetup.createDFISellOrder('BTC', accountDFI, '037f9563f30c609b19fd435a19b8bde7d6db703012ba1aba72e9f42a87366d1941', new BigNumber(15), new BigNumber(0.01))
+    const { makeOfferTxId } = await icxSetup.createDFIBuyOffer(createOrderTxId, new BigNumber(0.10), accountBTC)
+    const { DFCHTLCTxId } = await icxSetup.createDFCHTLCForDFIBuyOffer(makeOfferTxId, new BigNumber(10), '957fc0fd643f605b2938e0631a61529fd70bd35b2162a21d978c41e5241a5220', 500)
+    await icxSetup.createExtHTLCForDFIBuyOffer(makeOfferTxId, new BigNumber(0.10), '957fc0fd643f605b2938e0631a61529fd70bd35b2162a21d978c41e5241a5220',
```

Comment:
hmm... i don't quite get this... is this necessary??

Reply:
yes. let me elaborate on this particular scenario
`maker` - want to sell 15 DFI to BTC, creates an order to sell 15 DFI
`taker `- want to buy 10 DFI from that order, makes an offer to buy 10 DFI
`maker` -Maker sees the offer, agrees to partially fill the order with 10 DFI buy offer above. Creates the DFC HTLC on DFC chain
`taker` - taker sees the DFC HTLC, verify the details, creates the EXT HTLC on BTC chain. Then submits the EXT HTLC details on the DFC chain using `icx_submitexthtlc` RPC.
`maker` - maker claims the EXT HTLC on BTC network by revealing the secret. 
`taker` - taker now sees the secret and claim DFC HTLC on DFC chain.

So to add to your original qustion, we need to have `submit ext htlc` step before you can claim the DFC HTLC

Reply:
may be i should rename the function `createExtHTLCForDFIBuyOffer ` to `submitExtHTLCForDFIBuyOffer `
will do that